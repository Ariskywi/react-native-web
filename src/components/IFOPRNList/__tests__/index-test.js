/* eslint-env jasmine, jest */

import React from 'react';
import IFOPRNList from '..';
import { mount, shallow } from 'enzyme';

describe('components/IFOPRNList', () => {
    test('renders simple list', () => {
        const component = shallow(
            <FlatList
                data={[{key: 'i1'}, {key: 'i2'}, {key: 'i3'}]}
                renderItem={({item}) => <item value={item.key} />}
            />
        );
        expect(component).toMatchSnapshot();
    });
    test('renders empty list', () => {
        const component = shallow(
            <FlatList data={[]} renderItem={({item}) => <item value={item.key} />} />,
        );
        expect(component).toMatchSnapshot();
    });
    test('renders null list', () => {
        const component = shallow(
            <FlatList
                data={undefined}
                renderItem={({item}) => <item value={item.key} />}
            />,
        );
        expect(component).toMatchSnapshot();
    });
    test('renders all the bells and whistles', () => {
        const component = shallow(
            <IFOPRNList
                ItemSeparatorComponent={() => <separator />}
                ListEmptyComponent={() => <empty />}
                ListFooterComponent={() => <footer />}
                ListHeaderComponent={() => <header />}
                data={new Array(5).fill().map((_, ii) => ({id: String(ii)}))}
                keyExtractor={(item, index) => item.id}
                getItemLayout={({index}) => ({length: 50, offset: index * 50})}
                numColumns={2}
                refreshing={false}
                onRefresh={jest.fn()}
                renderItem={({item}) => <item value={item.id} />}
            />,
        );
        expect(component).toMatchSnapshot();
    });
});
