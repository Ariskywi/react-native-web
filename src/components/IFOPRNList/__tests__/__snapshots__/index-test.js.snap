// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/FlatList renders all the bells and whistles 1`] = `
<VirtualizedList
  ItemSeparatorComponent={[Function]}
  ListEmptyComponent={[Function]}
  ListFooterComponent={[Function]}
  ListHeaderComponent={[Function]}
  data={
    Array [
      Object {
        "id": "0",
      },
      Object {
        "id": "1",
      },
      Object {
        "id": "2",
      },
      Object {
        "id": "3",
      },
      Object {
        "id": "4",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  getItemLayout={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={2}
  onEndReachedThreshold={2}
  onRefresh={[Function]}
  refreshing={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;

exports[`components/FlatList renders empty list 1`] = `
<VirtualizedList
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;

exports[`components/FlatList renders null list 1`] = `
<VirtualizedList
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;

exports[`components/FlatList renders simple list 1`] = `
<VirtualizedList
  data={
    Array [
      Object {
        "key": "i1",
      },
      Object {
        "key": "i2",
      },
      Object {
        "key": "i3",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
